selfservice/strategy/webauthn/settings.go

```diff
@@ -192,6 +192,7 @@ func (s *Strategy) continueSettingsFlowRemove(w http.ResponseWriter, r *http.Req
 	}
 	if len(updated) == 0 {
 		i.DeleteCredentialsType(identity.CredentialsTypeWebAuthn)
+		ctxUpdate.UpdateIdentity(i)
```

Comment:
Why did the credentials got removed in the wirst place here? Is it because ctxUpdate.Identity does not have the credentials set? If so, we should probably add a check in the update routine which checks that `ctxUpdate` has actually been touched. What do you think?

Reply:
Because `UpdateContext.GetIdentityToUpdate` returns identity without credentials which comes from session when `ctxUpdate.Update` is not used in a settings flow. As you said, I think we should it.
https://github.com/ory/kratos/blob/dd460db14aeca66324043ce46cb4eff674ec62f6/selfservice/flow/settings/strategy_helper.go#L41-L44

Reply:
I added check logic at [d0cd751](https://github.com/ory/kratos/pull/2233/commits/d0cd7512a1ceac555ba3e80d33c04c86c3ca27a5)!

Reply:
nice!